### PR TITLE
[1LP][RFR] Infra Provider Details view and necessary changes

### PR DESCRIPTION
**Pull request by izapolsk**

### CODE REVIEW

File: cfme/common/provider_views.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+from widgetastic.widget import View, Text
+
+from cfme import BaseLoggedInPage
+from widgetastic_patternfly import Dropdown
+from widgetastic_manageiq import DetailsToolBarViewSelector
+from widgetastic_manageiq import (BreadCrumb,
+                                  SummaryTable,
+                                  Button,
+                                  TimelinesView)
+from utils.version import current_version
```
Comment (mshriver) [inline cfme/common/provider_views.py]: Are we okay adding more utils.version.current_version uses when we're also actively trying to remove it?

**Comment by dajoRH** [comment]
Lint report for commit d7d8adefe7ccde6f421296d593b5ec7723b0176c:

`cfme/common/provider_views.py`:
- :red_circle: Line 11:1: **F401** *'widgetastic.utils.VersionPick' imported but unused*
- :red_circle: Line 11:1: **F401** *'widgetastic.utils.Version' imported but unused*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

File: cfme/common/provider_views.py (hunk 1 above)
Comment (izapolsk) [inline cfme/common/provider_views.py]: fixed

File: cfme/common/provider_views.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    title = Text('//div[@id="main-content"]//h1')
+    breadcrumb = BreadCrumb(locator='//ol[@class="breadcrumb"]')
+
+    @View.nested
+    class toolbar(ProviderDetailsToolBar):  # NOQA
+        pass
+
+    @View.nested
+    class contents(View):  # NOQA
+        pass
```
Comment (mshriver) [inline cfme/common/provider_views.py]: **Optional**
Comment here stating the view is provided by __getattribute__

File: cfme/common/provider_views.py [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+            if self.context['object'].appliance.version >= '5.7':
+                view_type = self.toolbar.view_selector.selected
+                if view_type == 'Summary View':
+                    return ProviderDetailsSummaryView(parent=self)
+
+                elif view_type == 'Dashboard View':
+                    return ProviderDetailsDashboardView(parent=self)
+
+                else:
+                    raise Exception('The form for provider with such name '
```
Comment (mshriver) [inline cfme/common/provider_views.py]: 'form' doesn't seem right in this context. '

Consider: 'The content view type "{<view_type>}" for provider "{<prov>}" does not exist'

File: cfme/common/provider_views.py (hunk 2 above)
Comment (izapolsk) [inline cfme/common/provider_views.py]: ah, you're right. I originally had such comments but forgot to move 
https://github.com/ManageIQ/integration_tests/pull/4387/files#diff-fa6a86dc0dc65ab876b17fc13b55f6b6R238

File: cfme/common/provider_views.py (hunk 3 above)
Comment (izapolsk) [inline cfme/common/provider_views.py]: fixed

File: cfme/common/provider_views.py [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+    """
+    monitoring = Dropdown(text='Monitoring')
+    configuration = Dropdown(text='Configuration')
+    reload = Button(title='Reload Current Display')
+    policy = Dropdown(text='Policy')
+    authentication = Dropdown(text='Authentication')
+
+    @View.nested
+    class view_selector(DetailsToolBarViewSelector):  # NOQA
+        pass
```
Comment (mfalesni) [inline cfme/common/provider_views.py]: You could possibly replace this with simpler `view_selector = View.nested(DetailsToolBarViewSelector)`, I believe this construciton should work too.

File: cfme/common/provider_views.py [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+    @View.nested
+    class contents(View):  # NOQA
+        # this is switchable view that gets replaced with concrete view.
+        # it gets changed according to currently chosen view type  every time
+        # when it is accessed
+        # it is provided provided by __getattribute__
+        pass
+
+    def __getattribute__(self, item):
+        # todo: to replace this code with switchable views asap
```
Comment (mfalesni) [inline cfme/common/provider_views.py]: Hehe, yes :)

File: cfme/common/provider_views.py [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+class ProviderDetailsView(BaseLoggedInPage):
+    """
+     main Details page
+    """
+    title = Text('//div[@id="main-content"]//h1')
+    breadcrumb = BreadCrumb(locator='//ol[@class="breadcrumb"]')
+
+    @View.nested
+    class toolbar(ProviderDetailsToolBar):  # NOQA
+        pass
```
Comment (mfalesni) [inline cfme/common/provider_views.py]: dtto like above.

File: cfme/common/provider_views.py (hunk 4 above)
Comment (izapolsk) [inline cfme/common/provider_views.py]: yeah, I just forgot to change it. rest of such calls are already changed to call View.nested instead of decorator

Comment (izapolsk) [inline cfme/common/provider_views.py]: or I did this in some of next prs...

File: cfme/common/provider_views.py (hunk 5 above)
Comment (izapolsk) [inline cfme/common/provider_views.py]: this is in my scope for this iteration :)

**Pull request merged**